sqlalchemy_sphinx/dialect.py

```diff
@@ -83,14 +97,16 @@ def visit_match_func(self, fn, **kw):
         if self.left_match and self.right_match:
             match_terms = []
             for left, right in zip(self.left_match, self.right_match):
-                t = "(@{} {})".format(self.process(left), right.value)
+                if left is None:
+                    t = u"{0}".format(self.dialect.escape_value(right.value))
+                else:
+                    t = u"(@{0} {1})".format(
+                        self.process(left),
+                        escape_special_chars(self.dialect.escape_value(right.value)))
                 match_terms.append(t)
             self.left_match = tuple()
             self.right_match = tuple()
-            return "MATCH('{}')".format(" ".join(match_terms))
-
-    def visit_distinct_func(self, func, **kw):
```

Comment:
What was the reasoning for removing the distinct function?

Reply:
SphinxQL supports only one form of distinct (`COUNT(DISTINCT something)`). In the other cases you should use `GROUP BY`.

Reply:
Hmmm unfortunately I don't work with Sphinx anymore; however, leaving in the visit distinct shouldn't cause any issues for now.